addons/website_slides/controllers/main.py

```diff
@@ -301,17 +301,40 @@ def create_slide(self, *args, **post):
     # --------------------------------------------------
     @http.route('/slides/embed/<int:slide_id>', type='http', auth='public', website=True, sitemap=False)
     def slides_embed(self, slide_id, page="1", **kw):
+        """
+        Detect if the slide is embedded in a website outside of the Odoo database.
+        The detection is there to differentiate cases where the slide is present on a webpage that is different than
+        the website for which the slide was created
+
+        This is how we decide if the slide is embedded or not:
+        if referer_url is Falsly -> Not embedded (it's the default behavior)
+
+        if there is a website_id (which means the slide is just available for 1 website) ->
+            is_embedded = False  if and only if  domain of referer is equal to domain of website
+        else (Available for all websites) ->
+            is_embedded = False  if and only if  domain of referer is equal to ANY of the domain of the odoo DB
+
+
+        This is important because externally embedding the slide will display Share options inside the slide itself,
+        rather than on the bottom of the page in the Odoo slide module
+
+        In case of uncertainty about the situation, we should assume it is NOT embedded
+        """
         # Note : don't use the 'model' in the route (use 'slide_id'), otherwise if public cannot access the embedded
         # slide, the error will be the website.403 page instead of the one of the website_slides.embed_slide.
         # Do not forget the rendering here will be displayed in the embedded iframe
 
         # determine if it is embedded from external web page
-        referrer_url = request.httprequest.headers.get('Referer', '')
-        base_url = request.env['ir.config_parameter'].sudo().get_param('web.base.url')
-        is_embedded = referrer_url and not bool(base_url in referrer_url) or False
-        # try accessing slide, and display to corresponding template
+        referrer_url = request.httprequest.headers.get('Referer', False)
+        referer_domain = tools.url_domain_extract(referrer_url)
         try:
             slide = request.env['slide.slide'].browse(slide_id)
+            website_id = slide.website_id
+            if website_id:
+                is_embedded = referer_domain and not referer_domain.endswith(website_id._get_http_domain())
+            else:
+                is_embedded = referer_domain and not any(referer_domain.endswith(web_id._get_http_domain()) for web_id
```

Comment:
if your test are green with this code, you can probably rewrite it...

referer_domain = odoo.com
web_id._get_http_domain() = https://www.odoo.com

so probably

`web_id._get_http_domain().endswith(referer_domain)
`
and referrer_url -> referer_url  ?

btw as explain by xmo, uses url_domain_extract is not correct...

Reply:
@JKE-be You're right for the referrer_url, I'll rename all of them to referer, I forgot about that

For the rest, if I understand correctly, the good solution for you would be
`domain = web_id._get_http_domain(); domain.endswith(foo(referer_url))` with `_get_http_domain()` returning a string like `https://www.xxxxxx.com`, and a special function `foo` (best candidate so far : werkzeug.urls.url_parse(referer_url).netloc or something similar) returning a string like `xxxxx.com` ?

But I am wondering
What if we have `domain = https://www.asite.odoo.com`, `referer_url = www.odoo.com`, and `foo(referer_url) = odoo.com`? Then it will be detected as not embedded even though it is

**However**, if we go with the other way around `foo(referer_url).endswith(domain)`, then that case will work, but a `domain = 'https://www.odoo.com`, `referer_url = www.asite.odoo.com`, and  `foo(referer_url) = asite.odoo.com` will be detected as not embedded even though it is

How should we avoid those issues you think ?